### PR TITLE
Define and use hkdf_extract.

### PR DESCRIPTION
**Pull request by jyao1**
Fix: https://github.com/DMTF/libspdm/issues/1102

Signed-off-by: Jiewen Yao <jiewen.yao@intel.com>

### CODE REVIEW

**Review by steven-bellock — CHANGES_REQUESTED** [review]

**Comment by steven-bellock** [comment]
I filed https://github.com/DMTF/Security-TF/issues/2133 on the specification side.

File: library/spdm_crypt_lib/libspdm_crypt_crypt.c [hunk 1]
```diff
@@ -1542,6 +1542,98 @@ bool libspdm_hmac_all(uint32_t base_hash_algo, const void *data,
...
+ * @param  base_hash_algo                 SPDM base_hash_algo
+ *
+ * @return HKDF extract function
+ **/
+libspdm_hkdf_extract_func get_spdm_hkdf_extract_func(uint32_t base_hash_algo)
+{
+    switch (base_hash_algo) {
+    case SPDM_ALGORITHMS_BASE_HASH_ALGO_TPM_ALG_SHA_256:
+#if LIBSPDM_SHA256_SUPPORT == 1
+        return libspdm_hkdf_sha256_extract;
```
Comment (steven-bellock) [inline library/spdm_crypt_lib/libspdm_crypt_crypt.c:1557]: Does `cryptlib.h` really need to implement, for example, `libspdm_hkdf_sha256_extract`? Why can't these functions live in `spdm_crypt_lib.h` and call, for example, `libspdm_hmac_sha256_all`?

Comment (jyao1) [reply library/spdm_crypt_lib/libspdm_crypt_crypt.c:1557]: We should follow the crypto API. And hmac will be called by the crypto lib API, such as https://github.com/Mbed-TLS/mbedtls/blob/f71e2878084126737cc39083e1e15afc459bd93d/library/hkdf.c#L49.

I don't see the need to call hmac directly.

Comment (steven-bellock) [reply library/spdm_crypt_lib/libspdm_crypt_crypt.c:1557]: Ok that'll be a separate issue on the dividing line between cryptography that libspdm manages and cryptography that `cryptlib.h` manages.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**